### DISCUSSION TITLE
Support for OverDrive desktop app ends

### DISCUSSION THREAD

**Discussion by ZolaLa9** [chbrown/overdrive#70] [General]
`This script will stop working when OverDrive finally decides to stop supporting the "classic" OverDrive app.`

It looks like the end is upon us. This notice has appeared on my public library's website: "The MP3 audiobook format will be removed and OverDrive's desktop app for Windows and Mac will stop being supported on November 13, 2024. Download audiobooks using the Libby app instead."

Thank you, @chbrown for sharing your script. It has been a godsend. If you find or create another solution, hope you will share ❤️

**Comment by cerinawithasea** [comment]
IM wondering if and how people are using libby

**Comment by ZolaLa9** [comment]
I've tried Libby. I don't particularly like it, but it may just be an issue of getting used to it. Blergh.

**Comment by raf-42** [comment]
I'm in the same boat. I did some digging, and at least as of right now (11/11/24), the method outlined [here](https://old.reddit.com/r/audiobooks/comments/12bqv3u/is_there_any_way_to_download_library_audiobooks/jeya6ee/) works. 

> Log in to Overdrive website from your computer
From Overdrive > Holds > Open Audiobook in Browser
Right-click page select inspect
In the Inspector click Network
Refresh the browser page containing the playing audiobook
Click the TOC and cue up each chapter
At this point you should see the MP3 files starting to appear in your Network inspector.
Right-Click one and select open in new tab
The MP3 file opens, click the three dots button and select download.

Basically each mp3 chapter seems to still be there. Now whether this will continue to work in the future, who knows.

**Comment by jdustinlewis** [comment]
Looks like .ODM downloading is officially gone. When trying to download a .ODM (using an agent switcher):

> OverDrive's desktop app for Windows and Mac is no longer supported. Download audiobooks using the Libby app instead.

Bummer! Thanks again to @chbrown for the excellent work on this script. It was a good run!

**Comment by equivocates** [comment]
Thanks @chbrown for this great script. Sad to see it come to an end like this.

**Comment by crakerjac** [comment]
ODM files aren't completely gone... Only the javascript to generate the download button has been removed....  This will likely stop working in the coming weeks after all ODM books/files have been returned.

![20241115_15h45m21s_grim](https://github.com/user-attachments/assets/77669c01-e5af-4349-938d-663a90e40774)

**Comment by crakerjac** [comment]
> > ODM files aren't completely gone... Only the javascript to generate the download button has been removed.... This will likely stop working in the coming weeks after all ODM books/files have been returned.
> 
> If these ODM files are still there (for now), is there a way to find and download them without the download button?

I followed [this](https://github.com/chbrown/overdrive/issues/64#issuecomment-2277014420) reply from a different issue thread.  As of now, it still generates an ODM download button.

**Comment by ZolaLa9** [comment]
An interesting comment on [Reddit](https://www.reddit.com/r/LibbyApp/comments/1hf0hgt/comment/m29s6fn/): 

> the library I work for has got overdrive to agree to maintain it [**mp3 downloads**] for special circumstances because for some patrons it’s an accessibility issue

Perhaps the workarounds will continue working ... indefinitely ...? :)

**Comment by Glocktober** [comment]
Check out the section "Obtain an ODM File"
[in https://github.com/brianpipa/OverdriveMP3Downloader](https://github.com/brianpipa/OverdriveMP3Downloader)

**Comment by bemehiser** [comment]
I wrote a Firefox plugin which fixes the button automatically.
https://github.com/bemehiser/overdrive-mp3-button

README updates in https://github.com/chbrown/overdrive/pull/67

**Comment by simonafield** [comment]
I wrote a Tampermonkey userscript that also provides the ability to bulk download (including ZIP file compile) and return your loans.

[Here it is.](https://github.com/simon-techkid/OverDrive.Userscript)

<img width="960" alt="example_extension" src="https://github.com/user-attachments/assets/bd4200c5-9bc3-496c-be9f-3dd70e20e068" />

**Comment by chbrown** [comment]
Thanks for the kind words and helpful tips :) 

I am closing this as covered by 42373c1 but for future reference I have also enabled ['Discussions'](https://github.com/chbrown/overdrive/discussions) for this repo, which is probably what this thread should have been in the first place.

**Comment by JaegerMaster** [comment]
Not able to download .ODM files today. Is this all over now?

**Comment by chbrown** [comment]
Oh nooooes! — same here :(

**Comment by brianpipa** [comment]
Yep - 404 now on the ODM URL. i think we are done.

**Comment by windsifter** [comment]
It was a wonderful ride!

Thank you so much for the adventure and everything I learned from the
countless books I got to listen to because of your work.

**Comment by bemehiser** [comment]
There is another thread talking about downloading files via the listen-in-browser page, using an audio downloader extension. Not seamless yet, but it looks like something promising to explore in the future, possibly for a browser plugin.
https://github.com/brianpipa/OverdriveMP3Downloader/issues/4#issuecomment-2629049449

**Comment by cerinawithasea** [comment]
here is what i shared in another thread im watching about the same stuff basically

> I am now using https://github.com/HeronErin/LibbyRip with a converter that i wrote https://github.com/cerinawithasea/libregrab-m4b-creator

i can not even believe i just shared my thing, please be kind i have never done this before ever and it was done out of sheer panic

**Comment by windsifter** [comment]
One could also explore here...
https://www.reddit.com/r/trackers/comments/1fj4u0i/myanonamouse_compared_to_other_trackers/

**Comment by JarrettR** [comment]
Wish I'd seen that yesterday before I rolled my own solution @cerinawithasea 

https://github.com/JarrettR/OverdriveUnspooler

**Comment by crakerjac** [comment]
@JarrettR Careful with that LibbyRip. I got ALL of my library cards banned with that tool. Not just the one card to used with LibbyRip but ALL the cards tied to Libby.

**Comment by AlpacaLips99** [comment]
Thanks @crackerjac. It seems to me that “be careful with” must mean “just don’t use it.”  I don’t see any way to use it carefully.

**Comment by bedge** [comment]
> [@JarrettR](https://github.com/JarrettR) Careful with that LibbyRip. I got ALL of my library cards banned with that tool. Not just the one card to used with LibbyRip but ALL the cards tied to Libby.

@crakerjac could you elaborate on your usage pattern that triggered this? Or, any additional insight you have on what triggered this? (with the benefit of hindsight)

I'm wondering if this was caused by "excessive" (in the eyes of some) use, or maybe a book a week type use.

**Reply by crakerjac** [reply]
> > [@JarrettR](https://github.com/JarrettR) Careful with that LibbyRip. I got ALL of my library cards banned with that tool. Not just the one card to used with LibbyRip but ALL the cards tied to Libby.
> 
> @crakerjac could you elaborate on your usage pattern that triggered this? Or, any additional insight you have on what triggered this? (with the benefit of hindsight)
> 
> I'm wondering if this was caused by "excessive" (in the eyes of some) use, or maybe a book a week type use.

I looked into this a bit more. It's not necessary related to the usage pattern. It's that the app notifies the Libby back end that a Javascript is being used to download the raw MP3 files. I believe Libby is using this indicator to ban library cards.
